### PR TITLE
Add support for configuring prefetch capacity

### PR DESCRIPTION
**Pull request by annawoodard**
This is a squashed cherry pick with a few changes removed
in order to just collect the changes related to prefetch capacity.
It is based on the following commits:

75f9d4eca8ec179564768c74b8c5cd67154ea1d8
5a221f2e8ad581b23e4b4b1ec86d3ee02e41e460
c89f3b2def16023bd08360705e923eeb73815013

### CODE REVIEW

File: parsl/executors/high_throughput/executor.py [hunk 1]
```diff
@@ -115,6 +115,11 @@ class HighThroughputExecutor(ParslExecutor, RepresentationMixin):
     max_workers : int
         Caps the number of workers launched by the manager. Default: infinity
 
+    prefetch_capacity : int
+        Number of tasks that could be prefetched over available worker capacity. Default:100.
```
Comment (ZhuozhaoLi) [inline parsl/executors/high_throughput/executor.py]: Default should be 0

File: parsl/executors/high_throughput/process_worker_pool.py [hunk 2]
```diff
@@ -48,9 +48,9 @@ class Manager(object):
     def __init__(self,
                  task_q_url="tcp://127.0.0.1:50097",
                  result_q_url="tcp://127.0.0.1:50098",
-                 max_queue_size=10,
                  cores_per_worker=1,
                  max_workers=float('inf'),
+                 prefetch_capacity=100,
```
Comment (ZhuozhaoLi) [inline parsl/executors/high_throughput/process_worker_pool.py]: 0 here.

File: parsl/executors/high_throughput/process_worker_pool.py [hunk 3]
```diff
@@ -72,6 +72,11 @@ def __init__(self,
              caps the maximum number of workers that can be launched.
              default: infinity
 
+        prefetch_capacity : int
+             Number of tasks that could be prefetched over available worker capacity. Default:100.
```
Comment (ZhuozhaoLi) [inline parsl/executors/high_throughput/process_worker_pool.py]: 0 here

File: parsl/executors/high_throughput/process_worker_pool.py [hunk 4]
```diff
@@ -461,6 +470,8 @@ def set_stream_logger(name='parsl', level=logging.DEBUG, format_string=None):
                         help="REQUIRED: ZMQ url for receiving tasks")
     parser.add_argument("--max_workers", default=float('inf'),
                         help="Caps the maximum workers that can be launched, default:infinity")
+    parser.add_argument("-p", "--prefetch_capacity", default=100,
```
Comment (ZhuozhaoLi) [inline parsl/executors/high_throughput/process_worker_pool.py]: 0 here

File: parsl/executors/high_throughput/executor.py (hunk 1 above)
Comment (annawoodard) [reply parsl/executors/high_throughput/executor.py]: fixed

File: parsl/executors/high_throughput/process_worker_pool.py (hunk 3 above)
Comment (annawoodard) [reply parsl/executors/high_throughput/process_worker_pool.py]: fixed

File: parsl/executors/high_throughput/process_worker_pool.py (hunk 4 above)
Comment (annawoodard) [reply parsl/executors/high_throughput/process_worker_pool.py]: fixed

File: parsl/executors/high_throughput/executor.py (hunk 1 above)
Comment (ZhuozhaoLi) [inline parsl/executors/high_throughput/executor.py]: Seems another 100 here.

Comment (annawoodard) [reply parsl/executors/high_throughput/executor.py]: good catch, fixed

File: parsl/executors/high_throughput/process_worker_pool.py [hunk 5]
```diff
@@ -129,6 +135,10 @@ def create_reg_message(self):
                'python_v': "{}.{}.{}".format(sys.version_info.major,
                                              sys.version_info.minor,
                                              sys.version_info.micro),
+               'block_id': self.block_id,
```
Comment (yadudoc) [inline parsl/executors/high_throughput/process_worker_pool.py]: `self.block_id` is not defined anywhere. I'll remove this.

**Comment by annawoodard** [comment]
Fixes #881

**Pull request merged**